### DISCUSSION TITLE
Question about extended characters

### DISCUSSION THREAD

**Discussion by boydkelly** [bytesparadise/libasciidoc#845] [General]
Hi,
Thanks for this project!   I'm just testing a few things, and since I am dealing with a lanugage with extended latin characters I noticed that libasciidoc converts them with these codes as such:
<p>M&#596;g&#596;k&#596;r&#596;baw ya kuman ka kan ka to denmins&#603;nw hakili la</p>

Whereas asciidoctor will preserve the charcters in the html output:
<p>Mɔgɔkɔrɔbaw ya kuman ka kan ka to denminsɛnw hakili la</p>

Is this something that can be tweaked with go on my end?  Thanks for any pointers!

**Comment by pjanx** [comment]
I've also noticed that libasciidoc converts everything non-ASCII into HTML entities, unlike Asciidoctor:
```
Путин — хуйло!
```
is transformed into:
```
<div class="paragraph">
<p>&#1055;&#1091;&#1090;&#1080;&#1085; &#8212; &#1093;&#1091;&#1081;&#1083;&#1086;!</p>
</div>
```
Note that libasciidoc already uses `<meta charset="utf-8">` in its template.

It may still be theoretically useful to "entitify" the output, though I wouldn't have this behaviour as the default, and it's something a post-processing filter can do, anyway, should such a need arise.

**Reply by xcoulon** [reply]
I'll see what I can do about it. Maybe there's call to `html.EscapeString` that we can get rid of. 
Could you open an issue about it, please?

**Reply by pjanx** [reply]
Done. `html.EscapeString` doesn't create entities like that.

**Reply by xcoulon** [reply]
cool, thanks for taking time to open an issue!

> html.EscapeString doesn't create entities like that.

oh, ok. I'll need to dig a little, then ;)

**Comment by pjanx** [comment]
The behaviour has been [aligned](https://github.com/bytesparadise/libasciidoc/commit/4dcf7c67f31faf0e83beb5fc35a49c6c277123ad) for 0.7.0.
